### DISCUSSION TITLE
How to generate process flow diagram static (SVG) images

### DISCUSSION THREAD

**Discussion by aspitarl** [IDAES/idaes-pse#1311] [Q&A] [answered]
Hi, I've developed a custom flowsheet and am interested in trying to generate some process flow diagram SVG images. I'm generally exploring ways to generate static images for a report. I think exporting a SVG file would be useful to then post-editing in, for example, Inkscape.

First, I am aware of the ideas-ui extension. I have that working and it seems like a great tool, but using the 'export image' feature in there only has the option to save a PNG file.   I've found in the documentation of previous versions of ideas documentation [here](https://idaes-pse.readthedocs.io/en/1.9.0/user_guide/vis/index.html) that the flowsheet visualizer used to be able to save directly as SVG, or SVG could be exported from the web ui. I'm guessing this functionality has been temporarily removed, but not sure. In any case, being able to automate the generation of this SVG file outside of the ideas-ui interface would be nice.

Alternative to the flowsheet visualizer I have found some examples that use 'tagging' to edit an existing SVG (the `write_pfd` method for the [NGCC example](https://idaes-examples.readthedocs.io/en/latest/active/power_gen/ngcc/ngcc_doc.html?highlight=write_pfd)). The [subcritical power plant flowsheet](https://idaes-pse.readthedocs.io/en/stable/reference_guides/model_libraries/power_generation/flowsheets/subcritical_power_plant.html) also does this. It seems that the `svg_tag` method used in these examples needs an already created SVG file with some sort of common naming scheme to update that SVG file. Were these original SVG files (e.g. [plant_pfd.svg](https://github.com/IDAES/idaes-pse/blob/main/idaes/models_extra/power_generation/flowsheets/subcritical_power_plant/plant_pfd.svg)) created manually beforehand or is there some sort of automated process I'm missing? I opened the `plant_pfd.svg` file in Inkscape but the objects did not appear to be named in any particular way.

**Comment by weilongfei123** [comment]
Hello, I want to know how you learn IDAES platform. Is there any information you can share?

**Reply by aspitarl** [reply]
Reading the documentation, looking at the examples, then making my own flowsheet and unit model. Understanding Pyomo (which IDAES is built on top of) seems to be a good first step.

**Comment by CopyDemon** [comment, marked as answer]
Thank you for reaching out with your question regarding the generation of SVG images for process diagrams. I appreciate your detailed exploration of the available tools and methods.

I'm pleased to inform you that the functionality to export SVG files directly, which you've mentioned, is indeed something we have been working on. The good news is that this feature will be available in our next release. This update aims to provide more flexibility and efficiency in generating and editing SVG images for reports and other documentation purposes.

I hope this new functionality will greatly benefit your work, and we look forward to your feedback on its implementation. Stay tuned for our next release, and thank you for using our tool.
